core-java-modules/core-java-io-3/src/test/java/com/baeldung/writebytearray/WriteByteArrayUnitTest.java

```diff
@@ -0,0 +1,83 @@
+package com.baeldung.writebytearray;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.google.common.io.ByteSink;
+import com.google.common.io.MoreFiles;
+
+public class WriteByteArrayUnitTest {
+    private static byte[] dataForWriting;
+
+    @BeforeClass
+    public static void setup() throws IOException {
+        dataForWriting = Files.readAllBytes(Paths.get("src/test/resources/example-image.jpg"));
+    }
+
+    @Test
+    public void whenUsingFileOutputStream_thenByteArrayIsWritten() throws IOException {
+        try (FileOutputStream outputStream = new FileOutputStream("src/test/resources/example-fos.jpg")) {
+            outputStream.write(dataForWriting);
+            assertThat(new File("src/test/resources/example-fos.jpg")).hasBinaryContent(dataForWriting);
+        } catch (IOException ioe) {
```

Comment:
As the test throws `IOException` we don't need this catch block.. Similarly, if it just does `throw ioe` then it's a redundant `catch` anyway